File: src/facade/java/com/my/db/Connection.java
package src.facade.java.com.my.db;

public class Connection {
    private ResultSet resultSet;

    public void connect(String db) {
        System.out.println("Connected to " + db);
    }

    public void executeStatement(Statement statement) {
        System.out.println("Executing statement " + statement.getStatement());
        resultSet = new ResultSet(statement.getStatement() + " result");
    }

    public ResultSet getResultSet() {
        return resultSet;
    }
}

File: src/facade/python/db/Connection.py
from src.decorator.facade.python.db.ResultSet import ResultSet
from src.decorator.facade.python.db.Statement import Statement


class Connection:
    result: ResultSet

    def connect(self, db: str):
        print("connected to " + db)

    def execute_statement(self, statement: Statement):
        print("Executing statement " + statement.getStatement())
        self.result = ResultSet(statement.getStatement())

    def get_result_set(self) -> ResultSet:
        return self.result
